### PR TITLE
build: apply Angular package format and `ng-packagr`

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

File: src/src/single-spa-angular.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const defaultOpts = {
   updateFunction: () => Promise.resolve(),
 };
 
-export default function singleSpaAngular(userOpts: SingleSpaAngularOpts): LifeCycles {
+export function singleSpaAngular(userOpts: SingleSpaAngularOpts): LifeCycles {
```
Comment (arturovt) [inline src/src/single-spa-angular.ts:18]: :warning: breaking change for `@4` because angular's package format and ng-packagr don't allow default exports.

Comment (joeldenning) [reply src/src/single-spa-angular.ts:18]: :'( that sucks

File: package.json [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
+    "build:schematics": "tsc -p tsconfig.schematics.json && run-s copy:schematics:*",
+    "copy:schematics:json": "cpx schematics/schematics.json lib/schematics",
+    "copy:schematics:schema": "cpx \"schematics/ng-add/schema*\" lib/schematics/ng-add",
+    "copy:schematics:files": "cpx \"schematics/ng-add/_files/**/**\" lib/schematics/ng-add/_files",
     "format": "prettier './**/*' --write",
     "// - INTEGRATION INSTALLS": "Install packages for integration apps",
     "install:integration:shop": "yarn --cwd integration/shop install",
     "install:integration:portal": "yarn --cwd integration/portal install",
-    "install:integration": "yarn install:integration:shop && yarn install:integration:portal",
+    "install:integration": "run-s install:integration:shop install:integration:portal",
```
Comment (joeldenning) [inline package.json:47]: Why sequential? Couldn't they run concurrently?

```suggestion
    "install:integration": "concurrently -n w: 'yarn:install:integration:*'",
```

File: package.json [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
+    "test": "yarn clean && yarn jest",
     "lint": "eslint ./src/**/*.ts",
     ":copy-schematic-files": "cpx \"src/**/_files/**/**\" lib",
     "parcel": "ngc -p tsconfig.parcel.json",
-    "spa": "tsc -p tsconfig.spa.json",
-    "build:webpack": "tsc -p tsconfig.node.json",
+    "// - SINGLE-SPA-ANGULAR": "Scripts for building single-spa-angular",
+    "build:single-spa-angular": "yarn ts-node tools/build",
+    "// - WEBPACK": "Scripts for building Webpack config transformer",
+    "prebuild:webpack": "rimraf lib/lib",
```
Comment (joeldenning) [inline package.json:35]: Why is there a lib/lib directory?

File: package.json [hunk 4]
```diff
@@ -94,12 +94,15 @@
...
     "lint-staged": "^10.1.3",
+    "ng-packagr": "^9.1.0",
+    "npm-run-all": "^4.1.5",
     "prettier": "^2.0.4",
     "rimraf": "^2.6.2",
     "rxjs": "~6.5.5",
     "serve": "^11.3.0",
     "start-server-and-test": "^1.10.11",
     "ts-jest": "^24.0.0",
+    "tsickle": "^0.38.1",
```
Comment (joeldenning) [inline package.json:105]: Why is this needed?

File: package.json [hunk 5]
```diff
@@ -1,28 +1,54 @@
...
+  },
+  "homepage": "https://github.com/single-spa/single-spa-angular#readme",
+  "private": true,
   "scripts": {
     "prepublishOnly": "yarn build && yarn test",
-    "build": "yarn clean && yarn build:webpack && yarn parcel && yarn spa && yarn :copy-schematic-files",
-    "clean": "rimraf lib parcel",
-    "test": "jest",
+    "clean": "rimraf lib pacel",
+    "build": "run-s clean build:single-spa-angular build:webpack build:schematics",
```
[the remark this answers is not in the document]
Comment (arturovt) [reply package.json:27]: Umm, yes. Basically `build:single-spa-angular` should be first, because it will create `lib` folder. Other scripts can be parallel.

File: package.json (hunk 3 above)
Comment (arturovt) [reply package.json:35]: webpack is emitted there

File: package.json (hunk 4 above)
Comment (arturovt) [reply package.json:105]: it's a part of ng-packagr, it adds jsdoc annotations for the closure compiler.

File: package.json [hunk 6]
```diff
@@ -1,28 +1,54 @@
 {
   "name": "single-spa-angular",
-  "version": "3.3.0",
-  "description": "Helpers for building single-spa applications which use Angular 2",
-  "main": "lib/browser-lib/single-spa-angular.js",
-  "schematics": "./schematics.json",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (arturovt) [reply package.json:3]: https://github.com/ng-packagr/ng-packagr/blob/master/docs/DESIGN.md#folder-layout

Not monorepo. Basically the root package.json we use for such stuff as scripts, dependencies and husky/prettier.

There is another `package.json` inside the `src` folder that is taken by ng-packagr. ng-packagr automatically adds webpack fields like `main/module/es2015/esm5` and `typings` field.

File: package.json (hunk 4 above)
Comment (joeldenning) [reply package.json:105]: If it's part of ng-packagr, then why do we need to install it separately from ng-packagr?

File: package.json (hunk 3 above)
Comment (joeldenning) [reply package.json:35]: Why then is the import for webpack at `single-spa-angular/lib/webpack` instead of `single-spa-angular/lib/lib/webpack`?

https://github.com/single-spa/single-spa-angular/pull/167/files#diff-def93996c46300cb52885ed7c3584275R1

File: package.json (hunk 4 above)
Comment (arturovt) [reply package.json:105]: Because it's not a peer dependency of ng-packagr.

File: package.json (hunk 3 above)
Comment (joeldenning) [reply package.json:35]: This question is the main one I'm waiting on before reviewing again to approve/merge ^

Comment (arturovt) [reply package.json:35]: The content of the lib folder is copied into node_modules/single-spa-angular folder

**Comment by arturovt** [comment]
@joeldenning I would like you to pull changes and test them locally. That would be great!

**Comment by joeldenning** [comment]
I'll take a look tomorrow

**Comment by joeldenning** [comment]
I didn't get to this today. Will try for tomorrow. Today, tomorrow, and Thursday I am doing full-time consulting with a company. I recently quit my job to try to take my work on open source full time, and do consulting a few days a month to help pay the bills. This week happens to be the week where I'm doing that consulting. So I'll try to fit it in in the next couple days, but if not then I will definitely take a look at this by Friday

**Comment by joeldenning** [comment]
I am looking at this now. Have a few questions that I'll share in a PR with a few small changes. If you're available and have time to chat over Slack, let's talk in the [single-spa slack workspace](https://single-spa.slack.com/join/shared_invite/enQtODAwNTIyMzc4OTE1LWUxMTUwY2M1MTY0ZGMzOTUzMGNkMzI1NzRiYzYwOWM1MTEzZDM1NDAyNWM3ZmViOTAzZThkMDcwMWZmNTFmMWQ).

**Comment by arturovt** [comment]
@joeldenning I'm having a Friday release. I will be available tomorrow.

**Comment by joeldenning** [comment]
👍 sounds good - I will send a PR up with some small changes and also have a list of questions in it.
